src/vs/workbench/api/node/extHostDebugService.ts

```diff
@@ -252,8 +252,8 @@ export class ExtHostDebugService implements IExtHostDebugService, ExtHostDebugSe
 		return this._debugServiceProxy.$unregisterBreakpoints(ids, fids, dids);
 	}
 
-	public startDebugging(folder: vscode.WorkspaceFolder | undefined, nameOrConfig: string | vscode.DebugConfiguration, parentSession?: vscode.DebugSession): Promise<boolean> {
-		return this._debugServiceProxy.$startDebugging(folder ? folder.uri : undefined, nameOrConfig, parentSession ? parentSession.id : undefined);
+	public startDebugging(folder: vscode.WorkspaceFolder | undefined, nameOrConfig: string | vscode.DebugConfiguration, parentSession?: vscode.DebugSession, options?: vscode.DebugSessionOptions): Promise<boolean> {
+		return this._debugServiceProxy.$startDebugging(folder ? folder.uri : undefined, nameOrConfig, parentSession ? parentSession.id : undefined, options ? { repl: options.repl } : undefined);
```

Comment:
I do not understand why this `options ?` check is needed, why don't we just pass the `options` directly

Reply:
That's good and in line with https://github.com/microsoft/vscode/wiki/Extension-API-guidelines#copy-data. An extension is free to give you a "bigger" object than needed or, worst, one with cycling dependencies. Therefore we copy.

Reply:
Ok, makes sense. 
I will now finally read the Extensino API guidelines.